port/resource_port_blueprint.go

```diff
@@ -460,28 +525,16 @@ func blueprintResourceToBody(d *schema.ResourceData) (*cli.Blueprint, error) {
 		if d, ok := p["description"]; ok && d != "" {
 			propFields.Description = d.(string)
 		}
-		switch propFields.Type {
-		case "string":
-			if d, ok := p["default"]; ok && d.(string) != "" {
-				propFields.Default = d.(string)
-			}
-		case "number":
-			if d, ok := p["default"]; ok && d.(string) != "" {
-				defaultNum, err := strconv.ParseInt(d.(string), 10, 0)
-				if err != nil {
-					return nil, err
-				}
-				propFields.Default = defaultNum
-			}
-		case "boolean":
-			if d, ok := p["default"]; ok && d.(string) != "" {
-				defaultBool, err := strconv.ParseBool(d.(string))
-				if err != nil {
-					return nil, err
-				}
-				propFields.Default = defaultBool
-			}
-		case "array":
+
+		df, defaultOk := p["default"]
+		dv, defaultValueOk := p["default_value"]
+		di, defaultItemsOk := p["default_items"]
+
+		if defaultOk && defaultValueOk && len(di.([]interface{})) != 0 && defaultItemsOk && len(dv.(map[string]interface{})) != 0 && df != "" {
```

Comment:
try to replace all the checks with ConflictsWith in the schema between "default", "default_value", "default_items"